src/v/storage/segment_appender.h

```diff
@@ -162,21 +205,76 @@ class segment_appender {
     // still heavy weight operations compared to regular flush()
     ss::future<> hard_flush();
 
+    enum write_state { QUEUED = 1, DISPATCHED, DONE };
```

Comment:
Any reason not to make this an enum class? Personally prefer them as a default.

Reply:
just muscle memory from the dark ages of c++, let me change it

Reply:
this is an enum class in the recent push